### ISSUE TITLE
deploy fails with "failed to create valid credentials: graphrbac.ApplicationsClient#Create"

### ISSUE THREAD

**Issue by seanknox** [Azure/acs-engine#1811]
**What version of acs-engine?**:
```
master:
commit 9ca92cbe31324dbd2c02093ae97536613b0c1d00 (HEAD -> master)
Author: Alexander Gabert <alex@xoreaxeax.de>
Date:   Tue Nov 21 17:51:31 2017 +0100

    typo intersted (#1808)

```
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:
- provision a Kubernetes cluster with RBAC enabled
- don't specify a service principal (let the CLI autocreate it)
```
acs-engine deploy --subscription-id ... --dns-prefix trestles-centralus --location centralus --auto-suffix --api-model  /Users/seanknox/go/src/github.com/Azure/acs-engine/examples/classic/kubernetes.classic.rbac.json
```

**What you expected to happen**:
Kubernetes cluster provisioned with RBAC enabled

**What Happened**:
Got:
` ServicePrincipalProfile was empty, and we failed to create valid credentials: "graphrbac.ApplicationsClient#Create: Failure`
```
$ acs-engine deploy --subscription-id e766d9ee-d3d9-4b63-a912-8963dcfdf655 --dns-prefix trestles-centralus --location centralus --auto-suffix --api-model  /Users/seanknox/go/src/github.com/Azure/acs-engine/examples/classic/kubernetes.classic.rbac.json
WARN[0001] To sign in, use a web browser to open the page https://aka.ms/devicelogin and enter the code BNXCHAC8H to authenticate.
WARN[0023] apimodel: missing masterProfile.dnsPrefix will use "trestles-centralus-5a149c68"
WARN[0023] --resource-group was not specified. Using the DNS prefix from the apimodel as the resource group name: trestles-centralus-5a149c68
WARN[0026] apimodel: ServicePrincipalProfile was missing or empty, creating application...
FATA[0097] apimodel invalid: ServicePrincipalProfile was empty, and we failed to create valid credentials: "graphrbac.ApplicationsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code=\"Unknown\" Message=\"Unknown service error\""
```

**Anything else**
Deploy worked the second time when I provided an existing SP.

**Comment by tariq1890** [comment]
I get the same issue when I try deploying with the example kubernetes apimodel json (which has an empty ServicePrincipalProfile). I tried simulating the same request via azure-sdk-for-go and I get the same error. It might be worth asking the SDK team to throw an error that provides for information. Thoughts?

**Comment by tariq1890** [comment]
This issue might be possibly related https://github.com/Azure/azure-sdk-for-go/issues/1587

**Comment by tariq1890** [comment]
As recommended by Joel, I updated the go-autorest dependency and I got a more informative error

`graphrbac.ApplicationsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Unknown" Message="Unknown service error" Details=[{"odata.error":{"code":"Request_BadRequest","date":"2018-07-10T05:53:21","message":{"lang":"en","value":"Another object with the same value for property identifierUris already exists."},"requestId":"a87de7e1-5377-47f1-b3ec-b0211aa963db","values":[{"item":"PropertyName","value":"identifierUris"},{"item":"PropertyErrorCode","value":"ObjectConflict"}]}}]`

Based on this error, I tried remedying the situation by suffixing the app name to create a different identifierURI and I get this error which indicates lack of permissions

`authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client '<redacted>' with object id '52d4e8f3-47cf-47a9-baa2-db6139d140d2' does not have authorization to perform action 'Microsoft.Authorization/roleAssignments/write' over scope '/subscriptions/<redacted>/resourceGroups/acs_deploy/providers/Microsoft.Authorization/roleAssignments/54ea37e2-6048-4dcd-86d9-a1e387fa10a6'."`

I guess the only way forward is to update the go-autorest dependency to a later version and get the exact error.
